blog/2024-01-23-state-machines-whats-in-a-name/index.mdx

```diff
@@ -0,0 +1,186 @@
+2. **Events/transitions**: Usually verbs
+   - Events are triggers that cause state transitions. They are actions or occurrences, so they are best named with verbs. Examples include **`submit`**, **`cancel`**, **`fail`**, **`load`**, **`sign in`**, or **`clear form data`**.
+3. **Actions**: verbs or verb phrases
+   - Actions are operations executed in response to events or state transitions. They should be verbs or verb phrases that describe what action is being taken. For instance, **`sendRequest`**, **`displayError`**, **`updateData`**.
+4. **Guards**: Often adjectives or past participles
+   - Guards are conditional checks that determine if a transition should occur. They can be named with adjectives or past participles that describe the condition being met. Examples include **`isLoggedIn`**, **`hasValidData`**, **`isExpired`**. You’ll notice these are similar to how you might name boolean variables in code.
+5. **Parent states**: Compound Nouns
+   - For nested states, these can also be nouns like regular states or they could be adjectives indicating a subordinate level of the parent state which is a noun. An example might be a state called **`Viewing Shopping Cart`** with child states **`Empty`** and **`Populated`**.
+6. **Invoked Actors**: nouns or gerunds
+   - When a state machine invokes a service (like an API call), these are often named with nouns or gerunds (nouns derived from verbs and ending in -ing) that describe the service's function. Examples are **`fetchingData`**, **`userAuthentication`**.
+
+By aligning the names in state machines with these parts of speech, the structure and functionality of the state machine becomes more intuitive and easier to understand, facilitating better communication and documentation.
+
+### Be brief yet specific
+
+Just like with naming anything in code, it’s helpful to pick names that are brief, not generic, yet contain just enough information to be unambiguous. Make names “minimally meaningful”.
+
+State name examples:
+
+- **`Idle`** instead of **`State1`**
+- **`Door Closed`** instead of **`Closed`**
+- **`Payment Processing`** instead of **`State_Awaiting_Payment`**
+
+Event name examples:
+
+- **`Submit`** or **`submit`** instead of **`Event1`**
+- **`Receive payment`** instead of **`Event_Payment_Received`**
+
+## What style should that name be in?
+
+When statecharts were first introduced in [“Statecharts: A visual formalism for complex systems” (Harel, 1987)](https://www.inf.ed.ac.uk/teaching/courses/seoc/2005_2006/resources/statecharts.pdf), there was relatively little discussion of naming conventions and many of the diagrams used letters and numbers to represetn states and events. The innovations that statecharts introduced were focused on visually representing relationships needed to support real world software use cases such as hierarchy, parallelism, history states, etc.
+
+<p>
+  <ThemedImage
```

Comment:
If both your light and dark images are the same, you don’t need the `themedImage` component, you can just use a markdown image `![alt text](url)` format.

Reply:
Thanks for the reminder about that!